flight/Modules/ManualControl/transmitter_control.c

```diff
@@ -490,139 +478,205 @@ static void set_armed_if_changed(uint8_t new_arm) {
 }
 
 /**
+ * Check sticks to determine if they are in the arming position
+ */
+static bool arming_position(ManualControlCommandData * cmd, ManualControlSettingsData * settings) {
+
+	bool lowThrottle = cmd->Throttle <= 0;
+
+	switch(settings->Arming) {
+	case MANUALCONTROLSETTINGS_ARMING_ALWAYSDISARMED:
+		return false;
+	case MANUALCONTROLSETTINGS_ARMING_ALWAYSARMED:
+		return true;
+	case MANUALCONTROLSETTINGS_ARMING_ROLLLEFT:
+		return lowThrottle && cmd->Roll < -ARMED_THRESHOLD;
+	case MANUALCONTROLSETTINGS_ARMING_ROLLRIGHT:
+		return lowThrottle && cmd->Roll > ARMED_THRESHOLD;
+	case MANUALCONTROLSETTINGS_ARMING_YAWLEFT:
+		return lowThrottle && cmd->Yaw < -ARMED_THRESHOLD;
+	case MANUALCONTROLSETTINGS_ARMING_YAWRIGHT:
+		return lowThrottle && cmd->Yaw > ARMED_THRESHOLD;
+	case MANUALCONTROLSETTINGS_ARMING_CORNERS:
+		return lowThrottle && (
+			(cmd->Yaw > ARMED_THRESHOLD || cmd->Yaw < -ARMED_THRESHOLD) &&
+			(cmd->Roll > ARMED_THRESHOLD || cmd->Roll < -ARMED_THRESHOLD) );
```

Comment:
What about Pitch? I think we should also require pitch down, so it is the same as on DJI controllers.